modules/listener/listener.bal

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/lang.array;
+import ballerinax/googleapis_gmail as gmail;
+
+# Listener for Gmail Connector.   
+public class Listener {
+    private string startHistoryId = "";
+    private string userId = ME;
+    private json requestBody;
+    private gmail:Client gmailClient;
+    private http:Listener httpListener;
+
+    public isolated function init(int port, gmail:Client gmailClient, string topicName) {
+        self.httpListener = checkpanic new (port);
```

Comment:
what is the reason for `checkpanic`?

Reply:
Earlier we have used `checkpanic` within Init. but now we replace this with `check` as a best practice. This is resolved here : https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail/pull/88/commits/baf0283abc43fe5cd9c2cb4da5d2a27c5f92c573#diff-63a1d463e07d11ddde62c1159ff6e6bcc5feb453c0c869510ca83da6dda1ae25R41